File: frontend/types/backgroundTask.ts
export interface BackgroundTask {
    id: string;
    task_id: string;
    status: string;
    retry_count: number;
    max_retries: number;
    error_message?: string;
    result_data?: Record<string, any>;
    started_at?: string;
    completed_at?: string;
    created_at: string;
    updated_at: string;
}

export interface BackgroundTaskCreate {
    task_id: string;
    status: string;
    retry_count: number;
    max_retries: number;
}

export interface BackgroundTaskUpdate {
    status?: string;
    error_message?: string;
    result_data?: Record<string, any>;
    started_at?: string;
    completed_at?: string;
}

export interface BackgroundTaskOption {
    value: string;
    label: string;
    status: string;
    retry_count: number;
    created_at: string;
}

File: backend/app/schemas/background_task.py
from pydantic import BaseModel
from typing import Optional, Dict, Any
from datetime import datetime
from uuid import UUID


class BackgroundTaskBase(BaseModel):
    task_id: str
    agency_id: UUID
    params: Dict[str, Any]
    status: str = "pending"
    retry_count: int = 0
    max_retries: int = 3


class BackgroundTaskCreate(BackgroundTaskBase):
    pass


class BackgroundTaskOut(BackgroundTaskBase):
    id: UUID
    error_message: Optional[str] = None
    result_data: Optional[Dict[str, Any]] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

